### PR TITLE
REF: pip-based testing/docs - use pyproject-specified extras

### PR DESCRIPTION
**Pull request by klauer**
* Use `pip`-provided extras for testing and documentation building
* Remove support for installing deps from text files
* Remove default CI extras for installation

### CODE REVIEW

**Comment by ZLLentz** [comment]
I think this is the correct approach
should I point to this branch in my pytmc PR to see what happens?

File: .github/workflows/python-docs.yml [hunk 1]
```diff
@@ -52,11 +52,6 @@ on:
         description: "Extra packages to be installed for documentation"
```
Comment (ZLLentz) [inline .github/workflows/python-docs.yml:52]: Should we drop these defaults?

**Review by tangkong — COMMENTED** [review]
So to make sure I understand where this landed w.r.t. changes in all our other repos, we need to:
* add to `pyproject.toml` (if it doesn't exist)
  * the `[tool.setuptools.dynamic.optional-dependencies.doc]` section pointing to `docs-requirements.txt`
  * the `[tool.setuptools.dynamic.optoinal-dependencies.test]` section pointing to `dev-requirements.txt`
* add `pytest-cov` to `dev-requirements.txt`
* add `docs-version-menu` to `docs-requirements.txt`
* pin `sphinx_rtd_theme>=1.2.0`  (not related to this PR directly, but necessary)

did I miss anything?

**Comment by klauer** [comment]
> pin `sphinx>=1.2.0` (not related to this PR directly, but necessary)

Zach pinned `sphinx_rtd_theme>=1.2.0` instead of sphinx

I think that summary is accurate (and hopefully complete) otherwise

**Comment by ZLLentz** [comment]
I'm going to do one more iteration on the pytmc pr and the twincat test pr to make sure those still run

**Comment by ZLLentz** [comment]
Adding to the above changes, in pyproject.toml:
```
[tool.pytest.ini_options]
addopts = "--cov=."
```

**Comment by ZLLentz** [comment]
the builds on the test branches all pass so I think we should merge this and switch the pytmc branch back to master

**Review by ZLLentz — APPROVED** [review]

**Comment by ZLLentz** [comment]
Ugh actually I think I tested those on an out of date branch, I'm going to have to update the pcdshub copy of the branch and run the fork tests again

**Comment by ZLLentz** [comment]
This did continue to work for the twincat build- that's the one I was worried about
pytmc passes too

**Comment by klauer** [comment]
A relief!

**Comment by ZLLentz** [comment]
apparently, this did break the docs-upload builds, which were only using the docs-extras to specify their needs
```
/home/runner/work/_temp/bceeccd3-3681-4d6d-9aab-f072046c815d.sh: line 14: docs-versions-menu: command not found
```
https://github.com/pcdshub/pytmc/actions/runs/5351058235/jobs/9704444179

Maybe that's a separate PR tomorrow, I can whip something up: if the docs-uploading is completely divorced from the docs generation, maybe the requirements should just be stated plainly in the build pipeline.

**Pull request merged**